pkg/handler/branch.go

```diff
@@ -35,12 +35,73 @@ func (h *branchHandler) GetActiveBranches(w http.ResponseWriter, r *http.Request
 	unit := queryParams.Get("unit")
 	number := queryParams.Get("number")
 
-	branches := []string{owner, repo, unit, number}
-	resp := GetActiveBranchesResp{
-		Branches: branches,
+	duration, err := time.ParseDuration(number + unit)
+	if err != nil {
+		http.Error(w, "Invalid time unit", http.StatusBadRequest)
+		return
+	}
+
+	var repoBranches []Branch
+	client := resty.New()
+	startTime := time.Now().Add(-duration)
+
+	apiURL := fmt.Sprintf("https://api.github.com/repos/%s/%s/branches", owner, repo)
+	resp, err := client.R().SetResult(&repoBranches).Get(apiURL)
+
+	if err != nil {
+		http.Error(w, "Failed to fetch branches from the GitHub API", http.StatusInternalServerError)
+		return
+	}
+
+	if resp.StatusCode() != http.StatusOK {
+		http.Error(w, "GitHub API returned status code: %d", resp.StatusCode())
+		return
+	}
+
+	var activeBranches []string
+	for _, branch := range repoBranches {
+		commit, _ := getCommit(owner, repo, branch.Commit.SHA)
+		latestCommitDate := commit.Author.Date
+		date, err := time.Parse(time.RFC3339, latestCommitDate)
+
+		if err != nil {
+			http.Error(w, "Failed to decode datetime", http.StatusInternalServerError)
+			return
+		}
+
+		if date.After(startTime) {
+			activeBranches = append(activeBranches, branch.Name)
+		}
 	}
 
-	_ = json.NewEncoder(w).Encode(resp)
+	response := GetActiveBranchesResp{
+		Branches: activeBranches,
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	_ = json.NewEncoder(w).Encode(response)
+}
+
+func getCommit(owner, repo, commitSHA string) (*Commit, error) {
+	var commit CommitDetails
+	client := resty.New()
+
+	resp, err := client.R().
+		SetResult(&commit).
+		Get(fmt.Sprintf("https://api.github.com/repos/%s/%s/commits/%s", owner, repo, commitSHA))
+
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode() != http.StatusOK {
+		return nil, fmt.Errorf("GitHub API returned status code: %d", resp.StatusCode())
+	}
+
+	var commitInfo = commit.Commit
+
+	return &commitInfo, nil
```

Comment:
Just curious - since we are not using **commitInfo** in the parent function, can we skip returning it from here?

Reply:
I am using this commit info in the parent function to find out if a branch has any commits within the specified period of time. Does that answer your question or was it something else?

Reply:
Oh sorry, I meant the error part. Shouldn't we read the error and return it it is not nil?


https://github.com/nyuoss/git-service/pull/10/files#diff-36d91272e285a092a7c5a5ac717cac1ec6a31f3e39f59fa1ac7bd3546f6a5c5dR63

Reply:
Right, thank you for the review, I have updated the same